src/sql/workbench/contrib/query/browser/queryActions.ts

```diff
@@ -640,8 +640,11 @@ export class ListDatabasesActionItem extends Disposable implements IActionViewIt
 
 	// EVENT HANDLERS FROM EDITOR //////////////////////////////////////////
 	public onConnected(): void {
-		let dbName = this.getCurrentDatabaseName();
-		this.updateConnection(dbName);
+		// Need to delay in order to wait for the current database to reassign itself if reconnecting via file explorer.
```

Comment:
And how do we know 200ms is enough? And what does it mean for the current database to "reassign itself"? We should only be getting this event when it's actually connected so if that isn't happening that seems like the bug we should be fixing instead.

Why is it creating a new connection when selecting from the file explorer anyways? If the file already has an editor opened then we should be finding that one and use the existing connection.

Reply:
I got the value from 200ms after doing tests with different values. 

What I mean by reassign itself is that when the function calls getCurrentDatabaseName, it tries to call getConnectionProfile on the connectionManagementService, which returns undefined (this behavior is only noticed when not debugging before that line, so this is a race condition)

syncConvertInput creates a new EditorInput when the file editor input is selected, and then calls connect with the profile that is retrieved, with new connectioncompletionoptions, this calls tryConnect, which then calls createNewConnection